### ISSUE TITLE
Wrong aws_transfer_user user_name validation

### ISSUE THREAD

**Issue by Skardian** [terraform-providers/terraform-provider-aws#8065]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

```
Terraform v0.11.7
+ provider.aws v2.3.0
```

### Affected Resource(s)

* aws_transfer_user

### Terraform Configuration Files

```hcl
resource "aws_transfer_user" "user" {
  server_id      = "s-0123456789abcdef0"
  user_name      = "user_name"
  role           = "arn:aws:iam::123456789012:role/transfer"
  home_directory = "/bucket/path"
}
```

### Expected Behavior
Terraform creates the user
```
 + aws_transfer_user.user
      id:             <computed>
      arn:            <computed>
      home_directory: "/bucket/path"
      role:           "arn:aws:iam::123456789012:role/transfer"
      server_id:      "s-0123456789abcdef0"
      user_name:      "user_name"
```

### Actual Behavior
Terraform fails to create users containing an underscore in the name
```
Error: aws_transfer_user.user: "user_name" isn't a valid transfer user name (only lowercase alphanumeric characters are allowed): "user_name"
```

### Steps to Reproduce

1. `terraform plan`

### References
AWS updated their user_name validation. Users with hypens and underscores are valid now.
![2019-03-25_11:10:21](https://user-images.githubusercontent.com/4048877/54911396-ae99aa80-4eee-11e9-8f3a-4c91b98ce17c.png)

* https://docs.aws.amazon.com/transfer/latest/userguide/API_CreateUser.html#SFTP-CreateUser-request-UserName

**Comment by nywilken** [comment]
Hi @Skardian thanks for reporting this issue. It looks like we just need to update the ValidateFunc for the username attribute. I’m going to mark this as an enhancement request. Please keep an eye on the issue for updates. 

https://github.com/terraform-providers/terraform-provider-aws/blob/06544e694b50c3d98060e50ecb009b17146ed289/aws/validators.go#L99

**Comment by ewbankkit** [comment]
[AWS SDK v1.19.3](https://github.com/aws/aws-sdk-go/releases/tag/v1.19.3): 

> This release also relaxes the SFTP user name requirements to allow underscores and hyphens.

**Comment by JukieWalsh** [comment]
PR opened #8304

**Comment by bflad** [comment]
Support for uppercase letter, hyphens, and underscores in `aws_transfer_user` user names to match the new AWS support has been merged and will release with version 2.8.0 of the Terraform AWS Provider, likely later this week.

**Comment by nywilken** [comment]
This has been released in [version 2.8.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#280-april-26-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
